File: exercises/ddd_infrastructure_layer/java/src/main/java/domain/application/PhoneNumber.java
package domain.application;
public class PhoneNumber {
    private final String value;

    public PhoneNumber(String value) {
        if (!isValidPhoneNumber(value)) {
            throw new IllegalArgumentException("Invalid phone number.");
        }
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    private boolean isValidPhoneNumber(String number) {
        // We would use more robust validation in a real application
        return number != null && number.matches("^\\d+$");
    }
}

File: exercises/ddd_infrastructure_layer/typescript/src/domain/application/PhoneNumber.ts
export class PhoneNumber {
  private readonly _value: string;

  get value() { return this._value; }

  constructor(value: string) {
    if (!this.isValidPhoneNumber(value)) {
      throw new Error(`Invalid phone number: ${value}`);
    }
    this._value = value;
  }

  private isValidPhoneNumber(value: string): boolean {
    // We would use more robust validation in a real application
    const phoneNumberPattern = /^\d+$/;
    return phoneNumberPattern.test(value);
  }
}
